File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/product/controller/CategoryController.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.product.controller;

import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.dto.CategoryResponse;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.service.CategoryService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@Slf4j
@RestController
@RequiredArgsConstructor
@RequestMapping("/api")
public class CategoryController {
    private final CategoryService categoryService;

    @GetMapping("/v1/categories")
    public ResponseEntity<List<CategoryResponse>> getAllCategories() {
        return ResponseEntity.ok(categoryService.getAllCategories());
    }
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/product/controller/CategoryController.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.product.controller

import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.dto.CategoryResponse
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.service.CategoryService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/categories")
class CategoryController(
    private val categoryService: CategoryService
) {

    @GetMapping("/")
    fun allCategories(): List<CategoryResponse> =
        categoryService.getAllCategories()
}
